collect_app/src/main/java/org/odk/collect/android/fragments/FormManagerList.java

```diff
@@ -160,6 +163,10 @@ public void onClick(DialogInterface dialog, int i) {
     private void deleteSelectedForms() {
         // only start if no other task is running
         if (backgroundTasks.deleteFormsTask == null) {
+            dialogFragment.setMessage(getResources().getString(R.string.form_delete_message));
+            dialogFragment.setCancelable(false);
+            dialogFragment.show(getActivity().getSupportFragmentManager(), ProgressDialogFragment.COLLECT_PROGRESS_DIALOG_TAG);
```

Comment:
Like in the other PR this should use `DialogUtils.showIfNotShowing` and `DialogUtils.dismissDialog`. That way you also don't need to hold a reference to the dialog like you do here. In fact, it's important to realize you shouldn't ever hold a reference to a `Fragment` instead. This is because if the Activity is recreated (like during a rotation) the Fragment will be recreated as well and the instance you have a reference to will not be the instance that is actually displayed. This causes a memory leak (holding a reference to an object that could have been garbage collected) and also might lead to bugs.